### PR TITLE
Getting-started with deel-lip

### PR DESCRIPTION
**Pull request by Sharing-Sam-Work**
I have created two new tutorial notebooks specifically tailored for professionals in industrial sectors who want to swiftly grasp the practical applications of the package. These tutorials offer a concise introduction to utilizing the package for producing and training robust 1-Lipschitz deep learning models. 

The focus here is on practical implementation rather than delving deeply into theoretical aspects. We provide practical suggestions to enhance user-friendliness and usability, making these tutorials ideal for those aiming for a practical working knowledge of the library and its functionalities, rather than an exhaustive theoretical exploration.

I have tested the changes with: tox -e py310-lint
I have also visualized the changes with: mkdocs serve

If the changes are validated, the google collab link will need to be changed in docs/index.md, so that they come from the deel-lip repository as opposed to mine (Sharing-Sam-Work), as in the below:
| Getting started 1 - Creating a 1-Lipschitz neural network  | [Open In Colab](https://colab.research.google.com/github/Sharing-Sam-Work/deel-lip/blob/master/docs/notebooks/Getting_started_1.ipynb)            |

### CODE REVIEW

**Review by cofri — COMMENTED** [review]
Very good introductory notebooks! I believe it makes things clearer and smoother for inexperienced users. These tutorials enhance the main tools of deel-lip: the custom layers and the specific losses.
I suggested some improvements. Feel free to take them into account or not. We can discuss about them later on.

**Comment by thib-s** [comment]
Thank you @Sharing-Sam-Work for this great work 👍 
In order to fully integrate this work we need to : 
- [ ] integrate @cofri comments
- [ ] add the badges pointing to the final url of the notebook
- [ ] rebase the branch and squash commits

My suggestion is to merge this PR in a temporary branch here in this repo so we can take care of this without requiring @Sharing-Sam-Work to do it.

File: docs/notebooks/Getting_started_1.ipynb [hunk 1]
```diff
@@ -0,0 +1,470 @@
...
+    "\n",
+    "## 📚 Theoretical background <a id='theoretical_background'></a> <a name='theoretical_background'></a>\n",
+    "### What is a Lipschitz constant\n",
+    "The `deel-lip` package allows to control the Lipschitz constant of a layer or of a whole neural network. The Lipschitz constant is a mathematical property of a function (in our context of work, a layer or a model) that characterizes how much the output of the function can change with respect to changes in its input. \n",
+    "\n",
+    "In mathematical terms, a function $f$ is Lipschitz continuous with a **Lipschitz constant K** or more simply **K-Lipschitz** if for any given pair of points $x_1,x_2$, $K$ provides a bound on the rate of change of $f$:  \n",
+    "\n",
+    "$$||f(x_1)-f(x_2)||\\leq K||x_1-x_2||.$$\n",
+    "\n",
+    "For instance, given a 1-Lipschitz dense layer (a.k.a fully connected layer) with a weight matrix $W$ and a bias vector $b$, we have for any two inputs x_1 and x_2: $$||W.x_1+b-(W.x_2+b)|| \\leq 1||x_1-x_2||.$$\n",
```
[the remark this answers is not in the document]
Comment (Sharing-Sam-Work) [reply docs/notebooks/Getting_started_1.ipynb:38]: Done

File: docs/notebooks/Getting_started_1.ipynb [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+    "\n",
+    "### A simple requirement for creating 1-Lipschitz neural network\n",
+    "The composition property of Lipschitz continuous functions states that if you have a function f that is K_1-Lipschitz and another function g that is K_2 Lispchitz, then their composition function h = (f o g) which applies f after g is also Lipschitz continuous with a Lipschitz constant K = K_1 * K_2.\n",
```
[the remark this answers is not in the document]
Comment (Sharing-Sam-Work) [reply docs/notebooks/Getting_started_1.ipynb:41]: Same here, I suppose that LaTeX dollars $ are missing around K_1, K_2 and K = K_1 * K_2. -> DONE

File: docs/notebooks/Getting_started_1.ipynb [hunk 3]
```diff
@@ -0,0 +1,470 @@
...
+    "\n",
+    "A neural network is essentially a stack of layers that transform the output of the layer before them and whose output is fed to the layer after them. \n",
+    "\n",
+    "By the composition property of Lipschitz functions, *it suffices for each of the n individual layers L_i of a neural network model to be 1-Lipschitz, for the whole model to be 1-Lipschitz*.\n",
```
[the remark this answers is not in the document]
Comment (Sharing-Sam-Work) [reply docs/notebooks/Getting_started_1.ipynb:45]: Done

File: docs/notebooks/Getting_started_2.ipynb [hunk 4]
```diff
@@ -0,0 +1,662 @@
...
+    "then:\n",
+    "$$y_{\\epsilon}=y.$$\n",
+    "\n",
+    "> \"For an input $x$, we can certify that are no adversarial perturbations constrained to be within a $\\text{L1}$-norm ball of certificate $\\epsilon_{x,\\text{L1}}$ that will change our model's prediction.\"\n",
+    "\n",
+    "For a given $x$, $\\forall \\epsilon$ such that $||\\epsilon||_{L1}<\\epsilon_{x,\\text{L1}}$, we obtain that:\n",
+    "$$A(x)=y,$$\n",
+    "$$A(x+\\epsilon)=y_{\\epsilon},$$\n",
+    "then:\n",
+    "$$y_{\\epsilon}=y.$$\n",
```
[the remark this answers is not in the document]
Comment (Sharing-Sam-Work) [reply docs/notebooks/Getting_started_2.ipynb:114]: Done:

Added: 🚨 Note: deel-lip only deals with L2 norm

Added in the first notebook:
💡 The norm we refer to throughout our notebooks is the Euclidean norm (L2). This is because deel-lip operates with this norm. You will find more information about the role of the norm in the context of adversarially robust 1-Lipschitz deep learning models in the notebook titled 'Getting Started 2'.

File: docs/notebooks/Getting_started_2.ipynb [hunk 5]
```diff
@@ -0,0 +1,662 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "359fa47e-86f3-440d-971f-0a49fe7dcdc7",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import deel\n",
```
[the remark this answers is not in the document]
Comment (Sharing-Sam-Work) [reply docs/notebooks/Getting_started_2.ipynb:182]: Done

File: docs/notebooks/Getting_started_2.ipynb [hunk 6]
```diff
@@ -0,0 +1,662 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "id": "df14bc65-3ebb-4f3c-925b-9a21cefa7e23",
+   "metadata": {},
+   "source": [
+    "We show two cases. In the first case, we use `deel-lip`'s `TauCategoricalCrossentropy` from the `losses` submodule. In the second case, we use another loss function from `deel-lip`: `MulticlassHKR`.\n",
+    "\n",
+    "In particular, we will show how these functions can be parametrized to increase the robustness of our predictive models. We will also see that generally, there is a compromise between the robustness and the accuracy of our models (i.e. better robustness generally comes at the price of a decrease in performance).\n",
+    "\n",
```
[the remark this answers is not in the document]
Comment (Sharing-Sam-Work) [reply docs/notebooks/Getting_started_2.ipynb:171]: Done.
Changed the content:

### 🎮 Control over the accuracy-robustness trade-off with `deel-lip`'s loss functions.

When training training 1-Lipschitz networks, one will see that there is a compromise between the robustness and the accuracy of the models. In simple terms, achieving stronger robustness often involves sacrificing some performance.

In this section, we will show the pivotal role of `deel-lip`'s loss functions in training 1-Lipschitz networks. Each of these functions comes with its own set of hyperparameters, enabling you to precisely navigate and adjust the balance between accuracy and robustness.

We show two cases. In the first case, we use `deel-lip`'s `TauCategoricalCrossentropy` from the `losses` submodule. In the second case, we use another loss function from `deel-lip`: `MulticlassHKR`.

File: docs/notebooks/Getting_started_2.ipynb [hunk 7]
```diff
@@ -0,0 +1,662 @@
...
+    "        \n",
+    "        lip.layers.SpectralDense(\n",
+    "                64,\n",
+    "                activation=activation(),\n",
+    "                use_bias=True,\n",
+    "                kernel_initializer=\"orthogonal\",\n",
+    "            ),\n",
+    "        \n",
+    "        lip.layers.SpectralDense(\n",
+    "                y_train.shape[-1], activation=None, use_bias=False, kernel_initializer=\"orthogonal\"\n",
```
[the remark this answers is not in the document]
Comment (Sharing-Sam-Work) [reply docs/notebooks/Getting_started_2.ipynb:229]: Done

**Comment by Sharing-Sam-Work** [comment]
I have taken into account the comments.

**Comment by cofri** [comment]
The two Getting Started notebooks were merged into master. I suggest to close this PR
